controller/src/beerocks/master/db/db.h

```diff
@@ -1122,19 +1122,22 @@ class db {
     /**
      * @brief Removes client with least timelife remaining from persistent db (with preference to disconnected clients).
      * 
+     * @param [in] client_to_skip A client mac that should not be selected as cadidate. This is to prevent currently added node as candidate.
      * @return true on success, otherwise false.
      */
-    bool remove_candidate_client();
+    bool remove_candidate_client(sMacAddr client_to_skip = beerocks::net::network_utils::ZERO_MAC);
 
     /**
      * @brief Returns the preferred client to be removed.
      * Preference is determined as follows:
      * - Prefer disconnected clients over connected ones.
      * - According to above, the client with least time left before aging.
-     *
+     
+     * @param [in] client_to_skip A client mac that should not be selected as cadidate. This is to prevent currently added node as candidate.
```

Comment:
`client_to_skip` sounds like a w/a
why not just add candidate_mac as a parameter 
`bool remove_candidate_client(const sMacAddr &candidate_client_mac);`

Reply:
we are actually not removing the client provided - but skipping it in the search of a candidate for removal